pkg/loop/internal/pb/ccipocr3/models.proto

```diff
@@ -4,22 +4,76 @@ option go_package = "github.com/smartcontractkit/chainlink-common/pkg/loop/inter
 
 package loop.internal.pb.ccipocr3;
 
-// CCIPMsg is a gRPC adapter to [github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3.CCIPMsg].
-message CCIPMsg {
-  uint64 sequence_number = 1;
-  BigInt chain_fee_limit = 2;
-  uint64 nonce = 3;
-  uint64 chain_fee_price = 4;
-  bytes message_id = 5; // Hash [32]byte
-  uint64 source_chain_selector = 6;
-  string sender = 7; // Address
-  string receiver = 8; // Address
-  bool strict = 9;
-  string fee_token = 10; // Address
-  BigInt fee_token_amount = 11;
-  bytes data = 12;
-  repeated TokenAmount token_amounts = 13;
-  repeated bytes source_token_data = 14;
+// RampMessageHeader is the family-agnostic header for OnRamp and OffRamp messages.
+// The MessageID is not expected to match MsgHash, since it may originate from a different
+// ramp family.
+message RampMessageHeader {
+  // MessageID is a unique identifier for the message, it should be unique across all chains.
+	// It is generated on the chain that the CCIP send is requested (i.e. the source chain of a message).
+  repeated bytes message_id = 1;
+  // SourceChainSelector is the chain selector of the chain that the message originated from.
+  uint64 source_chain_selector = 2;
+  // DestChainSelector is the chain selector of the chain that the message is destined for.
+  uint64 dest_chain_selector = 3;
+  // SequenceNumber is an auto-incrementing sequence number for the message.
+	// Not unique across lanes.
+  uint64 sequence_number = 4;
+  // Nonce is the nonce for this lane for this sender, not unique across senders/lanes
+  uint64 nonce = 5;
+  // MsgHash is the hash of all the message fields.
+  repeated bytes message_hash = 6;
+  // OnRamp is the address of the onramp that sent the message.
+  repeated bytes on_ramp = 7;
```

Comment:
This is an "optimization" of sorts because in order to hash the message we need the onramp address. This way the message hasher interface stays simple `Hash(ctx context.Context, msg cciptypes.Message)` instead of `Hash(ctx context.Context, msg cciptypes.Message, onRamp []byte)`.

The CCIP reader will create the Message structure offchain after reading from the chain directly, so it can fill this out because it should know the ramp its reading from.